docs/guides/collections.rst

```diff
@@ -0,0 +1,105 @@
+
+An entity object **must** take a collection as its **first** argument. This is to ensure that
+
+a) the entity can access an appliance through its collecion and
+b) that there is a consistent API for collection objects
+
+.. warning:: Objects **should** never be instantiated directly. They **should** always come from a
+             collection.
+
+
+Collection Methods
+------------------
+
+* ``__init__()`` - The collection object **must** take an ``appliance`` argument as its first argument
+  and assign this to the ``self.appliance`` attribute.
+* ``instantiate()`` - The collection object **should** provide an ``instantiate()`` method which
+  will simply return an entity instance with the supplied arguments. It **must** pass ``self``
+  as the first argument so tha tthe user doesn't have to.
+* ``create()`` - The collection object **should** provider a ``create()`` method where appropriate.
+  This method will attempt to create the object on the appliance and **must** then call
+  ``self.instantiate`` and return the object.
+
+.. note:: The ``__init__()`` method could in the future become part automatic in the BaseEntity class
+          but this is a future feature and is not yet planned.
+
+.. warning:: Failure to comply with the above guidelines in the future may result in an Exception
+             being raised
+
+Entity Methods
+--------------
+
+* ``__init__()`` - The entity object **must** take a ``collection`` argument as its first argument
+  and assign this to the ``self.collection`` attribute. It **must** then assign ``self.appliance`` to
+  be equal to ``self.collection.appliance``
+
+.. note:: The ``__init__()`` method could in the future become part automatic in the BaseEntity class
+          but this is a future feature and is not yet planned.
+
+.. warning:: Failure to comply with the above guidelines in the future may result in an Exception
+             being raised
+
+Example
+-------
+
+Below is an example of a generic object using the collection and entity relationships
+
+.. code-block:: python
+
+                from cfme.utils.appliance import BaseCollection, BaseEntity
+
+                class ObjectCollection(BaseCollection):
+                    """An object collection"""
+
+                    def __init__(self, appliance):
+                        self.appliance = appliance
+
+                    def instantiate(self, name, label):
+                        return Object(self, name, label)
+
+                    def create(self, name, label):
+                        run_create(name, label)
+                        return self.instantiate(name, label)
+
+                class Object(BaseEntity):
+                    """An object entity"""
+
+                    def __init__(self, collection, name, label):
+                        self.collecion = collection
+                        self.appliance = self.collection.appliance
+
+                    def update(self, *updates):
```

Comment:
Since you included this, consider importing utils.update.Updatable and adding it to class inheritence.

Reply:
I'm going to leave that for now, as I'm not sure what we wil do with update moving forward